performance/src/api_performance_tests.ts

```diff
@@ -1,87 +1,544 @@
-import { JSONObject, check, sleep } from 'k6';
-import { Options, Scenario, Threshold } from 'k6/options';
+import { JSONObject, check, sleep } from "k6";
+import { Options, Scenario, Threshold } from "k6/options";
 import http from "k6/http";
-import { jUnit, textSummary } from './utils/k6-summary';
-import { test, generateEndpointUrl } from './utils/helpers';
+import { jUnit, textSummary } from "./utils/k6-summary";
+import {
+    authenticateVU,
+    generateEndpointUrl,
+    test,
+    testGetRequest,
+} from "./utils/helpers";
 
-
-const DEFAULT_CLUSTER_URL = 'http://localhost:8080';
+const DEFAULT_CLUSTER_URL = "http://localhost:8080";
 
 // Fallback to localhost if a cluster url is not supplied
-const clusterURL = __ENV.DET_MASTER ?? DEFAULT_CLUSTER_URL
+const CLUSTER_URL = __ENV.DET_MASTER ?? DEFAULT_CLUSTER_URL;
+
+const RBAC_ENABLED = false;
+
+export const setup = (): TestConfiguration => {
+    const model = {
+        name: __ENV.model_name,
+        versionNum: __ENV.model_version_number,
+    };
+
+    const trial = {
+        id: __ENV.trial_id,
+    };
+
+    const workspace = {
+        id: __ENV.workspace_id ?? "1",
+        projectId: __ENV.project_id ?? "1",
+    };
+
+    const experiment = {
+        id: __ENV.experiment_id,
+        metricName: __ENV.metric_name,
+        metricType: __ENV.metric_type,
+        batches: __ENV.batches,
+        batchesMargin: __ENV.batches_margin,
+    };
+
+    const task = {
+        id: __ENV.task_id,
+    };
+
+    const seededData: SeededData = {
+        model,
+        trial,
+        experiment,
+        workspace,
+        task,
+    };
+
+    const token = authenticateVU(CLUSTER_URL);
+    const auth: Authorization = { token };
+    const testConfig: TestConfiguration = { auth, seededData };
+    getloadTests(testConfig, true);
+    return testConfig;
+};
 
 // List of tests
-const getloadTests = () => [
+const getloadTests = (
+    testConfig?: TestConfiguration,
+    inSetupPhase: boolean = false,
+): TestGroup[] => {
+    const sD = testConfig?.seededData;
 
-    // Query the master endpoint
-    test(
-        'visit master endpoint',
-        () => {
-            const res = http.get(generateEndpointUrl('/api/v1/master', clusterURL));
-            check(res, { '200 response': (r) => r.status == 200 });
-        }
-    ),
-]
+    const testSuite = [
+        // Query the master endpoint
+        test("get master configuration", () =>
+            testGetRequest("/api/v1/master", CLUSTER_URL, testConfig)),
```

Comment:
**nit suggestion:** one alternative to having to pass `CLUSTER_URL` and `testConfig` repetitively is to have a generative function approach:
```
export const testGetRequestor = (clusterUrl: string, testConfig?: TestConfiguration) => {
  return (url: string) => {
    const params = {
        headers: {
            "Content-Type": "application/json",
            Authorization: `${testConfig?.auth.token}`,
        },
    };
    const res = http.get(generateEndpointUrl(url, clusterURL), params);
    check(res, { "200 response": (r) => r.status == 200 });
  };
}
```
then you can just call:
```
const getRequest = testGetRequestor(CLUSTER_URL, testConfig);
...
getRequest("/api/v1/master");
...
```